paws_collector.js

```diff
@@ -440,6 +450,71 @@ class PawsCollector extends AlAwsCollector {
         this.reportDDMetric('api_throttling', 1)
         return cloudwatch.putMetricData(params, callback);
     };
+
+    /**
+     * Report the client errors and show case on DDMetrics and cloudwatch
+     * @param callback 
+     * @param error 
+     */
+    reportClientError(error, callback) {
+        var cloudwatch = new AWS.CloudWatch({ apiVersion: '2010-08-01' });
+        const params = {
+            MetricData: [
+                {
+                    MetricName: "PawsClientError",
+                    Dimensions: [
+                        {
+                            Name: 'CollectorType',
+                            Value: this._pawsCollectorType
+                        },
+                        {
+                            Name: 'FunctionName',
+                            Value: process.env.AWS_LAMBDA_FUNCTION_NAME
+                        }
+                    ],
+                    Timestamp: new Date(),
+                    Unit: 'Count',
+                    Value: 1
+                }
+            ],
+            Namespace: 'PawsCollectors'
+        };
+        let errorCode =  (typeof(error)=== 'object' &&  error.errorCode) ? error.errorCode:'unknown';
+        this.reportDDMetric("client", 1, [`result:error`,`error_code:${errorCode}`]);
```

Comment:
I would still call this "client_error" and distinguish it from successful client calls, rather than just relying on tags.

@daniel-greening what do you think?

Reply:
No real preference, though what's wrong with relying on tags?

Reply:
nothing I guess. I thought i would have been easier to build graphs off of. But that might just be my own way of doing things.